sklearn/_build_utils/__init__.py

```diff
@@ -70,20 +71,25 @@ def cythonize_extensions(extension):
         os.environ.get("SKLEARN_ENABLE_DEBUG_CYTHON_DIRECTIVES", "0") != "0"
     )
 
+    compiler_directives = {
+        "language_level": 3,
+        "boundscheck": cython_enable_debug_directives,
+        "wraparound": False,
+        "initializedcheck": False,
+        "nonecheck": False,
+        "cdivision": True,
+    }
+
+    if not Cython.__version__.startswith("0."):
+        compiler_directives["legacy_implicit_noexcept"] = True
```

Comment:
Given that this flag will be removed in the future, may we include a `TODO` comment here that links to https://github.com/cython/cython/issues/5088 and states that we need to fix this?